charts/generic-service/templates/ingress.yaml

```diff
@@ -14,15 +14,18 @@ metadata:
   {{- if .Values.allowlist }}
     nginx.ingress.kubernetes.io/whitelist-source-range: {{ include "app.joinListWithComma" .Values.allowlist | quote }}
   {{- end }}
-    external-dns.alpha.kubernetes.io/set-identifier: {{ $fullName }}-{{ .Release.Namespace }}-{{ .Values.ingress.contextColour }}
+    external-dns.alpha.kubernetes.io/set-identifier: {{ $fullName }}-{{ .Release.Namespace }}-green
+  {{- if not .Values.ingress.v1_2_enabled }}
+    kubernetes.io/ingress.class: {{- if .Values.ingress.modsecurity_enabled }} "modsec01" {{- else }} "nginx" {{- end }}
+  {{- end }}
   {{- if .Values.ingress.modsecurity_enabled }}
-    kubernetes.io/ingress.class: "modsec01"
     nginx.ingress.kubernetes.io/enable-modsecurity: "true"
     nginx.ingress.kubernetes.io/modsecurity-snippet: {{ tpl (toYaml .Values.ingress.modsecurity_snippet) . | nindent 6 }}
```

Comment:
Do we still need these?

Reply:
yes as mod security ingress controller is still an option

Reply:
Oh yeah! 😆